backend/local/watcher.coffee

```diff
@@ -122,17 +129,42 @@ class LocalWatcher
     onAdd: (filePath, stats) =>
         log.debug 'File added', filePath
         @paths?.push filePath
+        @pending[filePath]?.done()
```

Comment:
Well, it depends. Detecting moves/renames involves a lot of special cases. If I want to cover more of them, I have to add code that is not trivial and will have bugs. So, a balance is to be found.

In this case, if a file is deleted and then a file is added at the same path, I have to first marks the file as deleted in pouchdb before adding the file. Else, the file will be handled as a conflict. There are 2 solutions for that:

- mark the file as deleted right now ;
- defer the create event.

Defering the create event is very difficult to do right and needs a lot of code to support it. You have to take care of many things, like this add event can be related to a move, or defering it can cause deadlock (if you swap two files for example).

So, I've prefered the first solution for simplicity. The only drawback is that there is a short window of time where a move/rename can be not detected: if a file is renamed and just after that a file is created at the source location, it's possible that we will have 2 create and 1 delete events, instead of 1 create and 1 move.